### DISCUSSION TITLE
SMTP_DELIVERY_METHOD=sendmail with Exim fails

### DISCUSSION THREAD

**Discussion by Umiamz** [mastodon/mastodon#18304] [Q&A]
I'm running Mastodon 3.5.1 on Debian 11 using Exim4 as the MTA. Mail delivery is fine using localhost and port 25 for mail delivery options in .env.production, but if I set SMTP_DELIVERY_METHOD=sendmail, it fails. This is what is logged:

`May  2 23:31:50 mastodon bundle[21098]: Bad system call`

`May  2 23:31:50 mastodon kernel: [115565.189530] audit: type=1326 audit(1651530710.075:8): auid=4294967295 uid=1001 gid=1001 ses=4294967295 subj==unconfined pid=21099 comm="sendmail" exe="/usr/sbin/exim4" sig=31 arch=c000003e syscall=106 compat=0 ip=0x7fdb6b4d6782 code=0x80000000`

It looks like it's being blocked by kernel security but not sure if this is a Mastodon code issue or an Exim issue. Other applications don't appear to have any issues using sendmail-type mail delivery with Exim.

Is there any other logging I can look at to see if I can see what's happening, or has anybody else come across this?

Thanks for any input...

**Comment by Izorkin** [comment]
Try it with disabled AppArmor.

**Reply by Umiamz** [reply]
Thanks, but that doesn't make any difference - still fails with the same log entries.

**Reply by Izorkin** [reply]
If `audit: ` entries continue to appear, then AppArmor is not completely disabled

**Reply by Umiamz** [reply]
These look to be seccomp log entries rather than AppArmor ones.

**Reply by Izorkin** [reply]
So I got wrong. Sorry.
Try to disable sandboxing in mastodon-* services.

**Reply by Umiamz** [reply]
I've been experimenting with permitting various system calls and capabilities in mastodon-sidekiq.service and managed to make some progress but keep coming across yet another permission denied error in other parts of the Exim sending process. I think it's just easier to leave it using SMTP (localhost:25).

Oh well, I've learned something about systemd security tonight, anyway...
